src/kOS/UserIO/TelnetSingletonServer.cs

```diff
@@ -0,0 +1,1064 @@
+            {
+                // Note, for some infuriating reason, in my testing I often found that telnet clients ignored
+                // the fact that I am the server and I'm supposed to send the DO's while they are supposed to reply
+                // with the WILL or WONT codes.  Some clients (I'm looking at YOU, Putty) sometimes take on the 
+                // active side of the negotiation (the DO/DONT) and sometimes take on the passive side of it (the
+                // WILL/WONT responses to the DO's), mixed up all willy-nilly and not in adherence with the RFC's
+                // rules about which side does which part.  Therefore even though I JUST got done telling the client
+                // I'll do these things, it sometimes won't listen and demands that I respond to it's DO's instead.
+                // That's what this section of code is for:
+                
+                // If other side orders me to go char-at-a-time, agree or disagree depending on my setting:
+                // TODO: maybe some day alter this to obey the other side's wishes.
+                case RFC857_ECHO:
+                    if (isLineAtATime)
+                        SendTextRaw(new byte[] {RFC854_IAC, RFC854_WONT, RFC857_ECHO});
+                    else
+                        SendTextRaw(new byte[] {RFC854_IAC, RFC854_WILL, RFC857_ECHO});
+                    break;
+                // If other side orders me to go char-at-a-time, agree or disagree depending on my setting:
+                // TODO: maybe some day alter this to obey the other side's wishes.
+                case RFC858_SUPPRESS_GOAHEAD:
+                    if (isLineAtATime)
+                        SendTextRaw(new byte[] {RFC854_IAC, RFC854_WONT, RFC858_SUPPRESS_GOAHEAD});
+                    else
+                        SendTextRaw(new byte[] {RFC854_IAC, RFC854_WILL, RFC858_SUPPRESS_GOAHEAD});
+                    break;
+                default:
+                    offset += TelnetConsumeOther(RFC854_DONT, remainingBuff, offset);
+                    break;
+            }
+
+            // Everything below here is to help debug:
+            StringBuilder sb = new StringBuilder();
+            sb.Append("{"+RFC854_DO+"}");
+            sb.Append("{"+option+"}");
+            kOS.Safe.Utilities.Debug.Logger.SuperVerbose( "kOS: telnet protocol DO message from client: " + sb.ToString());
+
+            return offset;
+        }
+
+        
+        /// <summary>
+        /// Whenever an IAC WONT command is seen (see RFC 854), this reads through it and understands some parts of it.
+        /// </summary>
+        /// <param name="remainingBuff"> the buffer of raw stuff</param>
+        /// <param name="index">the offset into the buffer where the RFC654_WILL byte started</param>
+        /// <returns>how many bytes of the buffer should get skipped over because I dealt with them.</returns>
+        private int TelnetConsumeWont( byte[] remainingBuff, int index)
```

Comment:
Not yet but please don't delete it.  it's part of the protocol that I want to be in place to use later if and when we expand it.  The DO/DONT and WILL/WONT are parts of the RFC's for telnet and I just happen to have not made use of any of the options that have the server side expect a "wont" message from the client.  But if we expand this for things line line-at-a-time mode, then I'll need this and I wanted it here when I that part of the code and how to write it was still fresh in my head.